wremnants/helicity_utils.py

```diff
@@ -33,10 +33,17 @@ def makehelicityWeightHelper(is_w_like = False, filename=None):
         out = narf.ioutils.pickle_load_h5py(ff["results"])
 
     corrh = out["Z"] if is_w_like else out["W"]
+    if 'muRfact' in corrh.axes.name:
+        corrh = corrh[{'muRfact' : 1.j,}]
+    if 'muFfact' in corrh.axes.name:
+        corrh = corrh[{'muFfact' : 1.j,}]
+    missing = set(["y", "ptVgen", "chargeVgen", "helicity", "massVgen"]).difference(set(corrh.axes.name))
+    if missing != set():
+        raise ValueError (f"Axes {missing} are not present in the coeff histogram")
+    
     corrh = corrh.project('massVgen','y','ptVgen','chargeVgen', 'helicity')
```

Comment:
Is this this the expected order already? Why not just do a direct check on the order then rather than calling project?

```python
axes = ["y", "ptVgen", "chargeVgen", "helicity", "massVgen"]
if list(corrh.axes.name) != axes:
    if set(axes).difference(set(corrh.axes.name)).empty() == set():
         corrh = corrh.project(*axes) # reorder
    else:
        raise ValueError (f"Axes {axes} are not present in the coeff histogram")
```

Reply:
I think "list(corrh.axes.name) != axes" implies that you already need the correct sorting for it to work, two arrays are different even if their elements are the same but in scrambled order.
I imagine this was your goal here, but then  if list(corrh.axes.name) had more axes you may not catch it here

Reply:
True, should add an earlier check for the length. My point is that if the expectation is that the axes exist in that order, you might as well check for that and avoid the confusing and inefficient project

Reply:
ok, I see your point, I agree